### PR TITLE
#4 Add += and -= operators on Counter API

### PR DESCRIPTION
**Pull request by kkostov**
As proposed in #4, Extends the Counter API to enable increment and decrement of the counter's value using addition assignment and subtraction assignment operators e.g.

```swift
struct Schema: Codable, Equatable {
    var counter: Counter?
}

var doc1 = Document(Schema())
let _ = doc1.change {
    $0.counter?.set(3)
}

let _ = doc1.change {
    $0.counter -= 1
    XCTAssertEqual($0.counter?.get(), 2)
}

let _ = doc1.change {
    $0.counter += 1
    XCTAssertEqual($0.counter?.get(), 3)
}
```

### CODE REVIEW

File: Tests/Proxy+CounterTests.swift [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+//
+//  Created by Konstantin Kostov on 23/06/2022.
+//
+
+import Foundation
+import XCTest
+@testable import Automerge
+
+class ProxyCounterTests: XCTestCase {
+    func testAssignmentExpressions() {
```
Comment (lightsprint09) [inline Tests/Proxy+CounterTests.swift]: Can you pleas split this up into more isolated test cases.

**Review by lightsprint09 — CHANGES_REQUESTED** [review]
Looks fantastic. Thanks for bringing this up. If you split up the test cases I can merge it.

File: Tests/Proxy+CounterTests.swift (hunk 1 above)
Comment (kkostov) [reply Tests/Proxy+CounterTests.swift]: Thanks! Indeed it looks/reads much better as separate test cases.

**Review by lightsprint09 — APPROVED** [review]
Great. Thanks for splitting up the tests. Feel free to merge 👏🏾

**Comment by kkostov** [comment]
Awesome! I don't actually have permission to merge it by myself 😇

**Pull request merged**